include/IModule.hpp

```diff
@@ -9,6 +10,7 @@ class IModule {
     virtual ~IModule() = default;
     virtual auto update() -> void = 0;
     virtual operator Gtk::Widget &() = 0;
+  virtual void handleSeat(struct wl_seat*, uint32_t) {};
```

Comment:
I don't like this way of doing it, why can't we use the existing dp.emit when we get the seat?

Reply:
Yes, agreed. This is a RFC patch. I was just trying out to see if we can get the keyboard layout.
Could you please explain how we can use the dp.emit() call to send the seat details?

Reply:
You can store the seat in the bar and pass bar as parameters to the module like sway workspaces and so when emit is called in the module, you just need to access the variable

Reply:
I'll look into it!

Reply:
I searched for a solution. looks like this can be achieved only with a new protocol (or maybe there is an existing one?). 
The compositor currently sends modifier events only to the client which has the focus. The keyboard layout group is sent as part of the modifier event. We would need a new protocol to register for such modifier events so that we can change the layout based on the received event.

@Alexays Were you able to find something?

Reply:
I guess the way to go for now is polling then? Kind of sucks to have a delay.

Reply:
poll what? unless we receive the modifiers from the compositor we will never know of the layout change.

Reply:
There is one request in layer shell interface to set keyboard interactivity. Unfortunately, on setting it, all the keyboard events are sent only to the bar as it is the topmost :(

Reply:
> poll what? unless we receive the modifiers from the compositor we will never know of the layout change.

Ah, okay, my assumption was that the information about the current layout could be queried somewhere, but upon further reading it appears that it's managed by Sway and, right now, isn't exposed at all. Bummer.